src/main.js

```diff
@@ -0,0 +1,105 @@
+import core from '@actions/core'
+import tool from '@actions/tool-cache'
+import exec from '@actions/exec'
+import path from 'node:path'
+
+export async function updatecliExtract(downloadPath, downloadUrl) {
+  if (downloadUrl.endsWith('.tar.gz')) {
+    return tool.extractTar(downloadPath)
+  } else if (downloadUrl.endsWith('.zip')) {
+    return tool.extractZip(downloadPath)
+  } else {
+    throw new Error(`Unsupported archive type: ${downloadUrl}`)
+  }
+}
```

Comment:
FYI previous logic for `darwin` did not make sense as Xar is not supported on Mac 🤣

https://github.com/actions/toolkit/blob/c5278cdd088a8ed6a87dbd5c80d7c1ae03beb6e5/packages/tool-cache/src/tool-cache.ts#L297

Reply:
Thanks for the information.
As a none mac user, I just followed some documentation :).